models/issue.go

```diff
@@ -1061,17 +1064,13 @@ func newIssue(e *xorm.Session, doer *User, opts NewIssueOptions) (err error) {
 			return err
 		}
 		opts.Issue.Index = maxIndex + 1
-		if opts.Issue.Index == 0 {
-			return fmt.Errorf("Index numbers depleted")
-		}
-	} else {
-		perm, err := GetUserRepoPermission(opts.Repo, doer)
-		if err != nil {
-			return err
-		}
-		if !perm.IsAdmin() {
-			return ErrUserDoesNotHaveAccessToRepo{UserID: doer.ID, RepoName: opts.Repo.Name}
-		}
+	} else if !doer.IsAdmin {
```

Comment:
Not all issue created needs site admin, only when `opts.Issue.Index >0 && opts.Issue.Index < issueMaxAllowableIndex`.

Reply:
The way it's in my last commits:

* If it's a PR, Issue.Index comes from another part of the code and it is not validated (as it never was).
* If Issue.Index == 0, site admin is not required, but boundary is checked (i.e. no negative numbers are allowed).
* If Issue.Index !=0 and it's not a PR, then the user must be a site admin.

Is that adequate?

Reply:
I think you are right.

Reply:
OK. I think I did it right, then.